clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -557,6 +557,23 @@ bool implicitObjectParamIsLifetimeBound(const FunctionDecl *FD) {
   return isNormalAssignmentOperator(FD);
 }
 
+static bool hasLifetimeBoundAttribute(const Decl *D) {
+  if (D->hasAttr<LifetimeBoundAttr>())
+    return true;
+  if (const auto *AI = D->getAttr<LifetimeBoundIfAttr>()) {
+    bool Result;
+    // FIXME: we pay the cost of evaluating the binary condition everytime we
```

Comment:
Do we need to cache the evaluation or would it make sense to replace `LifetimeBoundIfAttr` with `LifetimeBoundAttr` once it evaluated to true so clients who already check for `LifetimeBoundAttr` does not need to be updated? Admittedly this would be a departure from what Clang AST is today, that is a really close representation of what the user wrote.